llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -2895,19 +2896,86 @@ Instruction *InstCombinerImpl::foldICmpSubConstant(ICmpInst &Cmp,
   return new ICmpInst(SwappedPred, Add, ConstantInt::get(Ty, ~C));
 }
 
+static Value *createLogicFromTable(const std::bitset<4> &Table, Value *Op0,
+                                   Value *Op1, IRBuilderBase &Builder) {
+  switch (Table.to_ulong()) {
+    case 0: // 0 0 0 0
```

Comment:
Please check again the following code and attach alive2 proof. I think you may have gotten the order of bits backward.